File: hard/2147_NumberofWaystoDivideaLongCorridor.py
class Solution:
    def numberOfWays(self, corridor: str) -> int:
        modulo = 10**9 + 7
        group1 = 1
        group2 = 0
        last = 0 
        for i,j in enumerate(corridor):
            if j == "S":
                group2 += 1
                if group2 > 2 and group2 % 2:
                    group1 = group1 * (i - last) % modulo
                last = i
        return group1 if group2 and group2 % 2 == 0 else 0
        

File: hard/2147_NumberofWaystoDivideaLongCorridor.js
/**
 * @param {string} corridor
 * @return {number}
 */
var numberOfWays = function(corridor) {
    const modul = 10**9 + 7
    let gr1 = 1
    let gr2 = 0
    let last = 0 

    for (let i = 0; i < corridor.length; i++){
        const j = corridor[i]
        if ( j == "S"){
            gr2 += 1
            if (gr2 > 2 && gr2 % 2){
                gr1 = gr1 * (i-last) % modul
            }
            last = i
        }
    }
    if (gr2 && gr2 % 2 == 0){
        return gr1
    }
    return 0
};